### ISSUE TITLE
Unstable formatting with parenthesized expressions

### ISSUE THREAD

**Issue by durin42** [psf/black#993]
Operating system: macOS 10.4.6
Python version: 3.7.2
*Black* version: c9689b80fbb108316b0330fc1ed444ccab419b02
Does also happen on master: Only happens on master.

This is a regression from 19.3b0, caught in the Mercurial code. I've extracted the relevant lines and modified one of them so that it triggers with an 88 column limit.

Input:
```
if True:
    if True:
        if True:
            if (group != gengroupall and
                (id in postfinalizegenerators) != (postfinalizeblah)):
                continue

if True:
    if True:
        if True:
            if True:
                if True:
                    errormsg = (_("push creates new branch '%s' "
                                  "with multiple heads") % (branch))
```

Error output:
```
--- source
+++ first pass
@@ -1,15 +1,17 @@
 if True:
     if True:
         if True:
-            if (group != gengroupall and
-                (id in postfinalizegenerators) != (postfinalizeblah)):
+            if group != gengroupall and (
+                id in postfinalizegenerators
+            ) != postfinalizeblah:
                 continue
 
 if True:
     if True:
         if True:
             if True:
                 if True:
-                    errormsg = (_("push creates new branch '%s' "
-                                  "with multiple heads") % (branch))
+                    errormsg = _(
+                        "push creates new branch '%s' " "with multiple heads"
+                    ) % branch
 
--- first pass
+++ second pass
@@ -1,17 +1,19 @@
 if True:
     if True:
         if True:
-            if group != gengroupall and (
-                id in postfinalizegenerators
-            ) != postfinalizeblah:
+            if (
+                group != gengroupall
+                and (id in postfinalizegenerators) != postfinalizeblah
+            ):
                 continue
 
 if True:
     if True:
         if True:
             if True:
                 if True:
-                    errormsg = _(
-                        "push creates new branch '%s' " "with multiple heads"
-                    ) % branch
+                    errormsg = (
+                        _("push creates new branch '%s' " "with multiple heads")
+                        % branch
+                    )
 
```

I think this might be related to #959, but it was different enough I went ahead and filed a bug.

**Comment by durin42** [comment]
`git bisect` implicates 957ba24bb6cdd32e4dd14ff2808dcf4d58851844 - I have to be done for the day, but hopefully that's helpful.

**Comment by durin42** [comment]
I noticed some fixes in this area have landed recently, but unfortunately this is still broken on mercurial/discovery.py and mercurial/transaction.py (as of c348 in our repo with line length limited to 80).

**Comment by durin42** [comment]
I suspect this was fixed by #1079 (but the hg code was already fixed).

**Comment by JelleZijlstra** [comment]
Yes, looks like it.
